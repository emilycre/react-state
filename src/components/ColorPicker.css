.ColorPicker {
  button {
    border: none;
    padding: 2em;
    margin: 1em;
    border-radius: 5px;

    cursor: pointer;
  }

  div {
    width: 125px;
    height: 125px;
    margin: 2em;
    border-radius: 50%;
  }

  .salmon {
    background-color: salmon;
    color: white;
  }

  .yellow {
    background-color: gold;
    color: gray;
  }

  .blue {
    background-color: teal;
    color: white;
  }
}
