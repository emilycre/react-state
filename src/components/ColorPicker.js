import React from 'react';
import PropTypes from 'prop-types';
import styles from './ColorDisplay.css';

function ColorPicker({ selectColorHandler }) {
  return (
    <section className={styles.ColorPicker}>
      <button className={styles.salmon} onClick={() => selectColorHandler('salmon')}>Salmon.</button>
      <button className={styles.yellow} onClick={() => selectColorHandler('yellow')}>Yellow.</button>
      <button className={styles.blue} onClick={() => selectColorHandler('blue')}>Blue-ish.</button>
    </section>
  );
}

ColorPicker.propTypes = {
  selectColorHandler: PropTypes.func.isRequired
};

export default ColorPicker;
