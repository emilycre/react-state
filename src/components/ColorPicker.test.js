import React from 'react';
import { shallow } from 'enzyme';
import ColorPicker from './ColorPicker';

describe('ColorPicker Component', () => {
  it('Renders a ColorPicker', () => {
    const wrapper = shallow(<ColorPicker />);
    expect(wrapper).toMatchSnapshot();
  });

  it('Changes color to salmon on button click', () => {
    const wrapper = shallow(<ColorPicker />);
    
    wrapper.find('button').at(0).simulate('click');
    expect(wrapper.state('color')).toEqual('salmon');
  });
});
