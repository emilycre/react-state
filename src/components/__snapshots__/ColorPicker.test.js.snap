// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ColorPicker Component Renders a ColorPicker 1`] = `
<section
  className="ColorPicker"
>
  <button
    className="salmon"
    onClick={[Function]}
  >
    Salmon.
  </button>
  <button
    className="yellow"
    onClick={[Function]}
  >
    Yellow.
  </button>
  <button
    className="blue"
    onClick={[Function]}
  >
    Blue-ish.
  </button>
  <div
    className=""
  />
</section>
`;
